### PR TITLE
Complex window solar or ground ray calculation potential overflow

### LINKED ISSUE #9415: Complex Window Solar Sky/Ground ray calculation possible float overflow
Issue overview
--------------
It was discovered that there could be a potential divide-by-zero for some scenarios in complex windows solar sky/ground calculation.
For a case using complex window on a skylight window, this will demonstrate a `WtSum` for the ground to be zero in the following code lines: 
https://github.com/NREL/EnergyPlus/blob/e95ef41afd4b0c7cbda25cb23701babafb50f7ad/src/EnergyPlus/WindowComplexManager.cc#L1533-L1538

This actually caused a divide-by-zero error for the debugger. However, the release version actually gets simulation finished, probably because since in this case `NGnd` is zero (which also causes `WtSum` to be zero), no actual value reassignment actually occurred since it gets an empty array range. But the float overflow error will actually occur.

A protection for divide-by-zero should be added for this ground part. Perhaps the same thing should be done for the sky part as well: 
https://github.com/NREL/EnergyPlus/blob/e95ef41afd4b0c7cbda25cb23701babafb50f7ad/src/EnergyPlus/WindowComplexManager.cc#L1531

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): Windows, possibly all
 - Version of EnergyPlus (if using an intermediate build, include SHA): 22.1
 - Discovered while working on helpdesk ticket 16172

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file [here](https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/9000s/9415))
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #9415
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by vidanovic** [comment]
This change seems simple and correct. I approve it.

**Review by vidanovic — APPROVED** [review]

File: src/EnergyPlus/WindowComplexManager.cc [hunk 1]
```diff
@@ -1538,14 +1538,22 @@ namespace WindowComplexManager {
             Geom.SolSkyWt(I) = SkyWeight(Geom.sInc(J));
         }
         WtSum = sum(Geom.SolSkyWt({1, NSky}));
-        Geom.SolSkyWt({1, NSky}) /= WtSum;
+        if (WtSum > DataGlobalConstants::rTinyValue) {
```
Comment (mjwitte) [inline src/EnergyPlus/WindowComplexManager.cc:1541]: This is a one-time setup process for each complex window, so some extra checks are not a computational issue.

File: tst/EnergyPlus/unit/WindowManager.unit.cc [hunk 2]
```diff
@@ -3121,3 +3122,76 @@ TEST_F(EnergyPlusFixture, WindowMaterialComplexShadeTest)
     EXPECT_NEAR(state->dataHeatBal->ComplexShade(1).SlatConductivity, 159.2276, 1e-5);
     EXPECT_NEAR(state->dataHeatBal->ComplexShade(1).SlatCurve, 0, 1e-5);
 }
+
+TEST_F(EnergyPlusFixture, SetupComplexWindowStateGeometry_Test)
+{
+    std::string const idf_objects = delimited_string({"WindowMaterial:ComplexShade,",
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/WindowManager.unit.cc:3128]: Nice concise unit test.

**Review by mjwitte — APPROVED** [review]
Ran unit tests locally.
New unit test fails with develop, passes here.
Defect file fails with develop (debug build), completes here (debug build).
CI is all green except the random SolarShadingTestGPU failure, and HospitalBaselineReheatReportEMS fails on the integration tests. Looks like a timeout, but we haven't seen that for a while, but there is no complex fenestration in this file.

@Myoldmopar Any comments before this merges?

**Comment by Myoldmopar** [comment]
Yeah no problem here, this looks like a reasonable change to protect against overflow, and a good unit test.  Merge at will.

**Pull request merged**